MdeModulePkg/Library/MemoryBucketLib/MemoryBucketLib.c

```diff
@@ -22,6 +22,13 @@ EFI_MEMORY_TYPE  mMemoryTypes[PEI_BUCKETS] = {
   EfiACPIMemoryNVS
 };
 
+enum {

```

Comment:
We already have an ENUM of memory types in include\Uefi\UefiMultiPhase.h --> EFI_MEMORY_TYPE.  Another enum with different values for the same thing shouldn't be created.

Reply:
Because we're not keeping track of all the memory types with the PEI memory buckets the enum in UefiMultiPhase doesn't match up with the indexes in the structures I created.  I originally hard coded numbers but it was recommended to me to make another enum to make the code more readable.

Reply:
Or, you could sparsely populate your arrays - it isn't that much memory.